src/ability/wasm.rs

```diff
@@ -0,0 +1,14 @@
+use libipld_core::{ipld::Ipld, link::Link};
+
+#[derive(Debug, Clone, PartialEq)]
+pub struct Run {
+    pub module: Module,
+    pub function: String,
+    pub args: Vec<Ipld>,
+}
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Module {
```

Comment:
@expede is more of a spec question. Can't this also be a URL? Or should there be a tag where it may be fetched from, as we do not always know it's a Cid from IPFS for example? So, there needs to be some info somewhere on how to retrieve it. Is it elsewhere and I'm not seeing it?

Reply:
> Can't this also be a URL?

There's no spec for the wasm ability yet, so we'll define it together!

The way I was thinking about this is that resolving a module for `$SOMEWHERE_ELSE` would be captured as an effect so that we can cache the output using the normal machinery. I think the higher level DSL will do things like construct the correct workflow for this. 

The other option is to special-case URL resolution specifically for Wasm at this layer, which I don't love — because we're already going to have that functionality in the system with our first-class effects — but I'm open to talking about it

Reply:
Yeah. We should discuss this, as it just needs to be defined somewhere. It's fine if it's outside of the execution (which is how it's done today in the runtime anyway but just pulled from what was `rsc` of course).